### DISCUSSION TITLE
高级设置工具需求收集

### DISCUSSION THREAD

**Discussion by zccrs** [linuxdeepin/developer-center#3236] [Features Request & Ideas | 特性请求 & 头脑风暴]
本贴**仅用于**为[高级设置工具](https://github.com/linuxdeepin/deepin-tweak)收集需求，将一直开放，为了保持帖子内容不偏离主题，请大家跟帖时遵循以下规则：

* 在回帖之前先阅读此项目的[README](https://github.com/linuxdeepin/deepin-tweak/blob/master/README_zh_CN.md)，了解它的定位和未来规划
* 只允许使用中文交流，如需使用英文，请移步到[这里]()
* 只允许发表功能建议，且是一个较为完整和确定的想法，不宜过于抽象，需让大部分人在阅读时都能较为容易的理解这段话的含义。尽可能使用陈述句，避免使用疑问句。条理清晰，叙事完整，在必要的情况下可适当配图说明
* 在回帖前浏览或搜索是否已经存在类似的建议，尽量做到不重复
* 所提的建议必须与此项目有关，且符合它的产品定位（即：提交到这里的建议应当是不可能添加控制中心，且没有希望在其它产品中改进的问题），所提出的建议不用考虑使用这个功能可能带来的负面影响，我们假定使用此工具的用户都有能力承担和解决使用它所带来的风险
* 遇到自己认同的需求建议，请为其打上`❤️`符号表示你有同样的需求，被“赞同”次数越多的建议越有可能被开发
* 遇到自己不认同的需求建议，可以添加`👎`符号表示不赞同他的观点，禁止回帖怼人
* 确定采纳的建议会被转为Issues单独跟踪，且会打上`🎉`符号，并在回帖中贴上它对应的Issues的链接

> **Warning** 这是对高级设置工具的需求征集帖！
> __！！！对其它产品的改进建议⚠不要⚠提到这里！！！__
> __对于违反规则的回帖，我们将直接删除，且不会提前进行通知！__
> __如果您对 deepin/DDE 或是此帖子本身有任何建议或想法，请开另外的帖子进行讨论，而不是在这个帖子中进行与需求收集无关的评论__

这是一个功能建议的例子：

````
问题：各类窗口的圆角大小不统一，deepin自己的应用默认都比较大（如控制中心），第三方的有边框的应用默认的比较小（如 firefox 浏览器），另外一些第三方无边框的应用则完全是直角（如 wps）。
建议一：可以增加一个“强制为所有窗口统一圆角大小”的选项，如果开启，在技术条件满足的情况下，无论窗口的类型，一律强制使用同样大小的圆角。
建议二：允许根据窗口名称匹配（可以是正则匹配），将匹配到的窗口添加到黑名单，对于黑名单里的窗口不强制设置圆角大小
````

__此外，帖子里的功能建议不一定都是官方人员开发，如果您是一名开发者，对其中的某个功能建议感兴趣，也可以在相关建议贴里回复自己的想法，并且 @BLumia 表示自己想要认领开发此功能。让我们来一起让它变得更强大吧！__

**Comment by zccrs** [comment]
问题：窗口标题栏太高了，比较占屏幕空间，本身一些软件自己顶部就需要比较多的空间，再加上窗口标题栏，能显示内容的空间大大减少。
建议一：增加一个设置窗口标题栏高度的功能，因为deepin自己的应用和使用系统边框的应用不一样，可以将两类窗口分别设置。

**Comment by zccrs** [comment]
问题：各类窗口的圆角大小不统一，deepin自己的应用默认都比较大（如控制中心），第三方的有边框的应用默认的比较小（如 firefox 浏览器），另外一些第三方无边框的应用则完全是直角（如 wps）。
建议一：可以增加一个“强制为所有窗口统一圆角大小”的选项，如果开启，在技术条件满足的情况下，无论窗口的类型，一律强制使用同样大小的圆角。
建议二：允许根据窗口名称匹配（可以是正则匹配），将匹配到的窗口添加到黑名单，对于黑名单里的窗口不强制设置圆角大小

**Reply by justforlxz** [reply]
直接在合成器里把边角裁切掉？

**Comment by zccrs** [comment]
问题：窗口特效不知道在什么情况下就无法打开了，在控制中心设置也无效。从KWin的代码中可以看出来，当窗管在特效模式下运行时，如果连续多次异常退出，将自动禁用特效模式，防止一直程序崩溃
建议一：添加一个强制开启特效模式的按钮，点击后将 ~/.config/kwinrc 中的 `OpenGLIsUnsafe=true`改成`OpenGLIsUnsafe=false`，再调用打开窗口特效的接口

**Comment by ZhongYic00** [comment]
问题：dock及控制中心等采用插件机制的应用只会在启动时扫描载入插件，开发调试时反复kill掉重启后易进入安全模式、导致start-dde对dock的自动重启失效
建议一：添加开发模式，开发模式下禁用DDE各组件崩溃检测及安全模式，联动start-dde对dock等进程的保活功能

**Reply by zccrs** [reply]
建议一理论上存在缺陷，加载个新的插件可以实现，但是如果这个插件是已经修改过了，现在修改了代码需要重新加载，那就必须重新启动控制中心或者是dock，因为没办法把已经加载过的旧插件完全卸载掉。

建议二可行，不过我建议改个形式，变成“添加一个开关，可以禁用DDE各组件的崩溃检测（如任务栏的安全模式），并且在关闭检测后自动退出安全模式，恢复到原始状态”

**Reply by zccrs** [reply]
如果你认同这个解释的话，可以直接修改一下这个建议的内容

**Reply by ZhongYic00** [reply]
目前来看直接kill也是一种不错的方法

**Reply by happyd0ggie** [reply]
题外话，针对建议一，是否可以定义一套接口标准，修改dock加载插件的机制，改为内核动态加载模块那样的机制？做到不重启dock重新加载插件。

**Reply by BLumia** [reply]
> 题外话，针对建议一，是否可以定义一套接口标准，修改dock加载插件的机制，改为内核动态加载模块那样的机制？做到不重启dock重新加载插件。

@happyd0ggie 建议另开 discussion 讨论

**Comment by tensor-tech** [comment]


**Reply by zccrs** [reply]
有哦，分别按 “Alt+空格”、“Alt + M”、“Alt + A” 快捷键就可以了

**Comment by ZhongYic00** [comment]
问题：音频设备有时候未关机就莫名其妙失效了，但不想关机/注销重启
建议：alsa仍能看到相应设备，推测只是控制中心后台失效，希望对音频、蓝牙等外设的后台服务添加手动重启

**Reply by kt286** [reply]
这个功能目前系统监视器里有

![image](https://user-images.githubusercontent.com/10684972/175845304-89e6e739-565b-4a3a-9550-8d4e7b39cd65.png)

**Comment by dempsey-wen** [comment]
问题：文件监听达到上限，导致在文件管理器中，新增或者删除或者重命名文件，都无法立即显示对应的修改，而必须要先离开当前路径的页面再回来（相当于刷新一下）才能看到对应的删改。

建议：添加一个修改文件监听上限的选项，输入[文件监听上限]，自动新建文件 /etc/sysctl.d/watches.conf，包含内容：fs.inotify.max_user_watches = [文件监听上限]

详情见该链接：https://bbs.deepin.org/zh/post/239197

**Comment by kt286** [comment]
问题：修改环境变量的时候，一直没搞明白要修改的文件。修改后可能还需要重启或注销才会生效。同时还有破坏配置文件的风险
建议一：可以增加一个“环境变量编辑”的功能，类似 Windows 中那样区分用户和系统。修改完成后实时生效，无需重启

**Reply by justforlxz** [reply]
/add TODO

NOTE: 无需重启只能针对修改后启动的程序。

**Reply by justforlxz** [reply]
https://github.com/linuxdeepin/deepin-tweak/pull/16 做了一个简单的，支持注销后生效 😭

**Reply by justforlxz** [reply]
> [linuxdeepin/deepin-tweak#16](https://github.com/linuxdeepin/deepin-tweak/pull/16) 做了一个简单的，支持注销后生效 😭

startdde 不支持临时设置环境变量，需要为 startdde 增加功能。

**Comment by justforlxz** [comment]
问题：没有一个地方可以展示用户的开机启动应用列表，启动器里的角标不是很明显
建议：提供开机启动项的列表，允许自行添加新的应用启动，或者删除某个应用。

**Comment by ghost** [comment]
问题：任务栏大小调整为非最小时，图标有灰色的圆角矩形背景。见[https://bbs.deepin.org/post/239628](https://bbs.deepin.org/post/239628)
建议：提供自由选择是否在图标后添加灰色背景的开关。

**Comment by linuxlionking** [comment]
问题：grub界面可定制化 美化程度不够高
建议：提供更灵活的定制功能以及美化功能

**Reply by ZanyRain** [reply]
可以通过命令行下载定制工具
sudo apt install grub-customizer

**Comment by shenmo7192** [comment]
建议：实现oh-my-dde现在已实现的功能

https://gitee.com/Limexb/oh-my-dde

* 热区设置

![图片](https://user-images.githubusercontent.com/47873776/190133649-b0894b7f-7587-4d77-bfb9-f63ea7ac3f48.png)

* 右键菜单管理

![图片](https://user-images.githubusercontent.com/47873776/190133791-edfa658f-f752-4a98-97e2-febe46cc4765.png)

**Comment by picsky** [comment]
问题：缩放、魔灯动画速度过快，显得十分生硬突然。使用systemsettings的话，又会存在潜在的风险。
建议：提供动画速度修改功能，就像systemsettings中的那样。

**Comment by happyd0ggie** [comment]
问题：通过gettings的方式设置键盘按键映射不方便，比如交换ctrl与caps，
建议：希望增加一个界面配置的入口。

**Comment by study-day** [comment]
深度定制要实现的东西，如果能可以不开特效的情况下实现，建议实现，比如反馈的大额头问题。
请认真考虑一下。如果由于技术原因实现不了，就委婉的说实现不了。请不要再删我的建议了。
这个建议我觉得可代表不开特效的群体的声音，不要装作听不见。


如果一开始没有特效模式那就没有这么多问题，如果存在就要维护起来。而且使用非特效模式的群体也不少，请重视。




罗琪
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "linuxdeepin/developer-center"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年9月27日(星期二) 下午5:16
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [linuxdeepin/developer-center] 高级设置工具需求收集 (Discussion #3236)





 通过gettings的方式设置键盘按键映射不方便，比如交换ctrl与caps，希望增加一个界面配置的入口。<br/&gt;<br/&gt;<br/&gt;<html&gt;
 <head&gt;
     <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"&gt;
 </head&gt;
 <body&gt;
 <div class="ntes-mailmaster-quote" style="padding-top: 1px; padding-bottom: 1px" &gt;
     <div style=" margin-top: 20px; margin-bottom: 12px; font-size: 14px; line-height: 1.25; color: #89898c; " &gt;---- 回复的原邮件 ----</div&gt;
     <div style=" margin-bottom: 12px; font-size: 13px; line-height: 1.25; color: #2c2c2e; padding: 2px; border-radius: 8px; background-color: #f0f0f0; " &gt;
         <table width="100%" cellpadding="0" cellspacing="10" border="0"&gt;
             <tr&gt;
                 <td valign="top" style=" width: 4em; font-size: 13px; line-height: 1.25; color: #89898c; white-space: nowrap; " &gt;发件人</td&gt;
                 <td valign="top" style=" font-size: 13px; line-height: 1.25; color: #2c2c2e; word-break: break-all; " &gt;<a class="mail-from" style="color: #1c83eb; text-decoration: none" ***@***.***" ***@***.***&amp;gt;</a&gt;</td&gt;
             </tr&gt;
             <tr&gt;
                 <td valign="top" style=" width: 4em; font-size: 13px; line-height: 1.25; color: #89898c; white-space: nowrap; " &gt;日期</td&gt;
                 <td class="mail-date" valign="top" style=" font-size: 13px; line-height: 1.25; color: #2c2c2e; word-break: break-all; " &gt;2022年09月19日 22:06</td&gt;
             </tr&gt;
             <tr style=""&gt;
                 <td valign="top" style=" width: 4em; font-size: 13px; line-height: 1.25; color: #89898c; " &gt;收件人</td&gt;
                 <td valign="top" style=" font-size: 13px; line-height: 1.25; color: #2c2c2e; word-break: break-all; " &gt;<a class="mail-to" style="color: #1c83eb; text-decoration: none" ***@***.***" ***@***.***&amp;gt;</a&gt;</td&gt;
             </tr&gt;
             <tr style=""&gt;
                 <td valign="top" style=" width: 4em; font-size: 13px; line-height: 1.25; color: #89898c; " &gt;抄送至</td&gt;
                 <td valign="top" style=" font-size: 13px; line-height: 1.25; color: #2c2c2e; word-break: break-all; " &gt;<a class="mail-cc" style="color: #1c83eb; text-decoration: none" ***@***.***" ***@***.***&amp;gt;</a&gt;、<a class="mail-cc" style="color: #1c83eb; text-decoration: none" ***@***.***" ***@***.***&amp;gt;</a&gt;</td&gt;
             </tr&gt;
             <tr&gt;
                 <td valign="top" style=" width: 4em; font-size: 13px; line-height: 1.25; color: #89898c; " &gt;主题</td&gt;
                 <td class="mail-subject" valign="top" style=" font-size: 13px; line-height: 1.25; color: #2c2c2e; word-break: break-all; " &gt;Re: [linuxdeepin/developer-center] 高级设置工具需求收集 (Discussion #3236)</td&gt;
             </tr&gt;
         </table&gt;
     </div&gt;
     <div&gt;<p&gt;</p&gt; 
 <p dir="auto"&gt;问题：动画速度过快，显得十分生硬突然。使用systemsettings的话，又会存在潜在的风险。<br&gt; 
 建议：提供动画速度修改功能，就像systemsettings中的那样。</p&gt; 
 
 <p style="font-size:small;-webkit-text-size-adjust:none;color:#666;"&gt;&amp;mdash;<br /&gt;Reply to this email directly, <a href="https://github.com/orgs/linuxdeepin/discussions/3236#discussioncomment-3681356"&gt;view it on GitHub</a&gt;, or <a href="https://github.com/notifications/unsubscribe-auth/ACNUZGRMPDGYGQ2Q77RKIT3V7BXNVANCNFSM5YJBABUQ"&gt;unsubscribe</a&gt;.<br /&gt;You are receiving this because you were mentioned.<img src="https://github.com/notifications/beacon/ACNUZGR2DI2T3DVD2GW4IZ3V7BXNVA5CNFSM5YJBABU2YY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOAA4CYTA.gif" height="1" width="1" alt="" /&gt;<span style="color: transparent; font-size: 0; display: none; visibility: hidden; overflow: hidden; opacity: 0; width: 0; height: 0; max-width: 0; max-height: 0; mso-hide: all"&gt;Message ID: <span&gt;&amp;lt;linuxdeepin/developer-center/repo-discussions/3236/comments/3681356</span&gt;<span&gt;@</span&gt;<span&gt;github</span&gt;<span&gt;.</span&gt;<span&gt;com&amp;gt;</span&gt;</span&gt;</p&gt; 
 <script type="application/ld+json"&gt;[ 
 { 
 ***@***.***": "http://schema.org", 
 ***@***.***": "EmailMessage", 
 "potentialAction": { 
 ***@***.***": "ViewAction", 
 "target": "https://github.com/orgs/linuxdeepin/discussions/3236#discussioncomment-3681356", 
 "url": "https://github.com/orgs/linuxdeepin/discussions/3236#discussioncomment-3681356", 
 "name": "View Discussion" 
 }, 
 "description": "View this Discussion on GitHub", 
 "publisher": { 
 ***@***.***": "Organization", 
 "name": "GitHub", 
 "url": "https://github.com" 
 } 
 } 
 ]</script&gt;</div&gt;
 </div&gt;
 </body&gt;
 </html&gt;

**Reply by justforlxz** [reply]
您好，本帖只负责收集高级设置工具的功能需求，不相关的讨论会被隐藏和删除。

我看了最开始的反馈，是希望要求不基于特效模式开发，实际上该程序在什么模式运行并不是程序自己控制的，高级设置工具是可以在非特效模式运行的。

高级设置工具只负责展示更多可调控的选项，不涉及具体的功能提供。最后重申一遍，该贴只负责收集需求，感谢您的支持。

**Reply by BLumia** [reply]
您好，请您 **仔细的** 阅读一下**这个帖子**的要求，这个帖子是 **高级设置工具** 的需求征集，**不是 deepin 本身** 的反馈帖子，对于后者请另开新的帖子进行反馈，而不是反馈到这个帖子里！

如果您不能做到按照帖子要求进行反馈的话，我们只能隐藏或折叠您的反馈，以确保其它的正常反馈不会被遗漏掉。

**Comment by study-day** [comment]
机器没有问题吧，那就是特效模式的问题了






罗琪
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "linuxdeepin/developer-center"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年10月17日(星期一) 中午1:14
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [linuxdeepin/developer-center] 高级设置工具需求收集 (Discussion #3236)





 
特效模式就是开启了窗口混成而已。还是卡的话要么是显卡驱动有问题，要么是机器实在不行。
 
这个回复和“高级设置工具”的主题本身似乎没什么关系，我就移除掉了。对于 deepin 的常规反馈建议另开新的主题帖来讨论。

**Comment by ice909** [comment]


**Reply by justforlxz** [reply]
这个工具是用来提供设置选项的，并不会提供额外的扩展组件扩展。

**Comment by xzl01** [comment]
问题:dde-daemon对于笔记本电源管理与其他第三方电源管理有冲突，如果用户想使用TLP等第三方电源管理，需要屏蔽dde-daemon的相关功能
建议：添加一个设置选项，调整dde-daemon相关行为，允许用户使用第三方电源管理软件

**Comment by xrui94** [comment]
建议增加以下两个设置项：
（1）增加时间和日期可以选择上下排列（即如下图1 Win11所示的排列方式）、左右排列（即目前V23的排列方式）
（2）增加是否显示图标背景色的设置（如下图2和图3）。即便状态图标是做了分组，但建议采用Win11上鼠标悬停时，才显示背景色的方案，而V23是一直显示背景色，这导致UI的割裂感很严重
![image](https://user-images.githubusercontent.com/29156275/224530789-0c6e7828-10df-4ceb-b237-45ae3f15fb0d.png)
![image](https://user-images.githubusercontent.com/29156275/224530806-1d852951-49a8-461e-968c-00fd805e0a4c.png)
![image](https://user-images.githubusercontent.com/29156275/224530816-ee5ed29e-558c-4cfe-914e-43c8c8bef43a.png)

**Comment by bigben446** [comment]
文件夹左侧的定制

来自于“新人第一次安装使用深度系统，几点体会几点建议”
https://bbs.deepin.org/post/255073
文件管理器打开后上面那几个像现在的win系统一样的视频、照片、文档、下载之类的常用文件夹，好学不学干嘛学它呢，不知道有没有办法把它们去掉。

**Reply by xzl01** [reply]
并不是这样 你可以看看别的系统 这个是一个Linux系统约定俗成的规范

**Reply by BLumia** [reply]
不确定文管目前是不是有按照 [xdg-user-dirs](http://www.freedesktop.org/wiki/Software/xdg-user-dirs/) 提供那些目录，如果有的话（先决条件），高级设置工具提供配置来管理 xdg-user-dirs 对应的配置也算可以接受。

**Reply by zccrs** [reply]
> 不确定文管目前是不是有按照 [xdg-user-dirs](http://www.freedesktop.org/wiki/Software/xdg-user-dirs/) 提供那些目录，如果有的话（先决条件），高级设置工具提供配置来管理 xdg-user-dirs 对应的配置也算可以接受。

实际上不太一样，即便用 xdg-use-dirs 修改了 user dirs，但是文管那里显示的还是这些“文档”、“音乐”等东西，只是它对应的目录变了。不过这是两码事了，在 deepin-tweak 中增加对 xdg user dirs 的设置我觉得是合适的。但是如果目标是为了文件管理器的侧边栏定制，那还得先靠文件管理器提供支持，而且如果它做了支持，最好是在文件管理器的界面上就能设置，没必要到 deepin-tewak 里了。

**Reply by BLumia** [reply]
我指的是从 xdg-use-dirs 对应的配置文件中 添加/删除 目录。当然这种影响相比是较大的，如果用户希望的只是隐藏的话，这种就不适合在 xdg-user-dirs 的配置文件里做了。

**Reply by zccrs** [reply]
> 我指的是从 xdg-use-dirs 对应的配置文件中 添加/删除 目录。当然这种影响相比是较大的，如果用户希望的只是隐藏的话，这种就不适合在 xdg-user-dirs 的配置文件里做了。

增删是不太好
